test/src/test/java/org/corfudb/integration/CorfuReplicationReconfigurationIT.java

```diff
@@ -143,6 +157,111 @@ public void testActiveClusterReset() throws Exception {
         // (6) Verify Data on Standby after Restart
         log.debug(">>> (6) Verify Data on Standby");
         verifyStandbyData((numWrites*2 + numWrites/2));
+
+        // (7) Verify replication status after all data has been replicated (no further data)
+        corfuStoreActive.openTable(LogReplicationMetadataManager.NAMESPACE,
+                LogReplicationMetadataManager.REPLICATION_STATUS_TABLE,
+                LogReplicationMetadata.ReplicationStatusKey.class,
+                LogReplicationMetadata.ReplicationStatusVal.class,
+                null,
+                TableOptions.fromProtoSchema(LogReplicationMetadata.ReplicationStatusVal.class));
+
+        // Wait the polling period time before verifying sync status (to make sure it was updated)
+        Sleep.sleepUninterruptibly(Duration.ofSeconds(LogReplicationAckReader.ACKED_TS_READ_INTERVAL_SECONDS + 1));
+
+        long remainingEntriesToSend = verifyReplicationStatus(ReplicationStatusVal.SyncType.LOG_ENTRY,
+                LogReplicationMetadata.SyncStatus.ONGOING, LogReplicationMetadata.SnapshotSyncInfo.SnapshotSyncType.DEFAULT,
+                LogReplicationMetadata.SyncStatus.COMPLETED);
+       assertThat(remainingEntriesToSend).isEqualTo(0L);
+
+        // (8) Keep writing data into the TX stream (but with data not intended for replication) while
+        // checking the status, confirm, remainingEntriesToSend is '0'
+        writerService.submit(() -> {
+            try {
+                openNonReplicatedTable();
+                while (!stopWrites.get()) {
+                    writeNonReplicatedTable(numWrites);
+                }
+            } catch (Exception e) {
+                fail("Failed Test!");
+            }
+        });
+
+        // Wait the polling period time and verify sync status again (to make sure it was not erroneously updated)
+        Sleep.sleepUninterruptibly(Duration.ofSeconds(LogReplicationAckReader.ACKED_TS_READ_INTERVAL_SECONDS + 1));
+
+        // While the TX log is growing, the remaining entries to send can be changing during this time, as it is computed
+        // wrt. the tail of the log and this varies depending on how fast we are catching the tail of the log.
+        verifyReplicationStatus(ReplicationStatusVal.SyncType.LOG_ENTRY,
+                LogReplicationMetadata.SyncStatus.ONGOING, LogReplicationMetadata.SnapshotSyncInfo.SnapshotSyncType.DEFAULT,
+                LogReplicationMetadata.SyncStatus.COMPLETED);
+
+        stopWrites.set(true);
+
```

Comment:
may be a good idea to verify remainingEntriesToSend == 0 here also

Reply:
No, if you see the comment in line 194 we can't ensure that because it is a running target, so likely you're always at least 1 behind or chasing the tail. So not being deterministic we don't know what to expect.